### PR TITLE
Newest: Fix auto post loading on every page visit

### PR DESCRIPTION
**Pull request by Gtes**
Main Page Newest  Posts FIX

## GitHub Board

**Issue link**

[#121 Fix auto post loading on every page visit ](https://github.com/ita-social-projects/dokazovi-fe/issues/121 )

## Summary of issue

- [x]  Check if we load posts for the first time.
- [x]  Either use separate actions or perform this check in the current action creator.

## Summary of change
- [x]  Added additional thunk for initial newest posts load

### CODE REVIEW

**Review by SergeOlabin — APPROVED** [review]

**Review by SergeOlabin — CHANGES_REQUESTED** [review]
pls resolve conflicts :)

File: src/modules/main/store/mainSlice.ts [hunk 1]
```diff
@@ -136,6 +136,19 @@ export const fetchExperts = createAsyncThunk('main/loadExperts', async () => {
...
 
+export const fetchInitialNewestPosts = (): AppThunkType => async (
+  dispatch,
+  getState,
+) => {
+  const {
+    main: { newest },
+  } = getState();
+
+  if (newest.newestPosts.length === 0) {
```
Comment (SergeOlabin) [inline src/modules/main/store/mainSlice.ts:147]:
```suggestion
  if (!newest.newestPosts.length) {
```
:)

**Review by SergeOlabin — APPROVED** [review]

File: src/modules/main/store/mainSlice.ts (hunk 1 above)
Comment (SergeOlabin) [reply src/modules/main/store/mainSlice.ts:147]: you can not fix that now, just letting you know that it's also possible

File: src/modules/main/components/NewestContainer.tsx [hunk 2]
```diff
@@ -24,7 +29,7 @@ const NewestContainer: React.FC = () => {
   });
 
   useEffect(() => {
-    setNewest();
+    setNewestInitial();
```
Comment (calster00) [inline src/modules/main/components/NewestContainer.tsx:32]: additional thunk for initial load is redundant, we can check if there are posts right here :)
```
    if (newestPosts.length === 0) {
      setNewest();
    }
```

**Review by calster00 — CHANGES_REQUESTED** [review]

File: src/modules/main/components/NewestContainer.tsx (hunk 2 above)
Comment (Gtes) [reply src/modules/main/components/NewestContainer.tsx:32]: On one of the syncs, we have discussed using the separate thunk for the initial load.

Checking length in the actual component was the first solution but was rejected.

**Pull request merged**